### DISCUSSION TITLE
how to use drm in plyr

### DISCUSSION THREAD

**Discussion by sinazahed** [sampotts/plyr#2415] [Q&A] [answered]
hi , is there a way to prevent user from video capture ?
i studied about drm in video player that show the user ip or name , email in the video player .
is it possible in plyr or any solution ?

**Comment by sadrian80** [comment, marked as answer]
Hi sinazahed

plyr.io does not have DRM capabilities on its own. 

But what you're describing are two different things:
DRM is a method that manages the interaction between the user and the content or the server delivering the content. So DRM might either apply to encrypting a video and decrypting it on the user-end or DRM might apply, if a user-request for a video has to call with specific credentials to identify with the server before content gets delivered.

Watermarking is the second thing you refer to. Many sites show information like user-id, timecode and ip-address above a video, so that a recording can be traced back. Watermarking can be done in different ways. The information about the user can be retrieved from the database (for example in Wordpress) and then be put above the video as a (transparent) layer. By animating the styling of it, you get a moving watermark. Most dynamically generated watermarks can be turned of with very simple measures by a user.

DRM is no 100% protection against theft. Watermarking even less so, as it can removed if it is not «burned» or «hardcoded» into the video stream. All measures do only prevent a certain segment of users, introducing hurdles for author and user along the way. My advice would be to plan carefully what measures you really need for the project and introduced only as much as needed.

As mentioned initially, plyr.io does not offer DRM on itself. Even when incorporating a DRM capable video host, like bunny.net, the player has work together with the CDN to manage DRM. Because of this, most service have their own player, like bunny.net uses a fork of plyr.io in their product.

Watermarking is also no native feature of plyr.io as far as I see. But implementing one is quite simple using PHP+CSS or JavaScript, depending on what you know better.

If you need a readymade solution, I would suggest you have a look at bunny.net Streaming. They have a 14 day trial and the service is really affordable. They offer DRM when you use their player, yet no watermarking.

**Reply by sinazahed** [reply]
Hi sadrian80
Thank you very much for your answer.
I have a another question about Plyr , is there a way to hide Plyr from wappalyzer ?
thank you so much

**Reply by sadrian80** [reply]
I can't possibly think of a positive reason to do that and I see no meaningful way to achieve that. Plyr is always visible in the source code of your site. JavaScript and jQuery run client-side. So I'd guess that you can't hide it directly from wappalyzer. There are possibilities to render the player outside of your actual code and embed it. But this too, makes it not impossible to detect but only a little less obvious.
